### ISSUE TITLE
Create Global Setup and Teardown Extensibility Points

### ISSUE THREAD

**Issue by RichiCoder1** [RichiCoder1/Browser.xUnit#1]
One of the bigger goals of this framework is to allow bootstrapping the environment to run browsers against, as well as enabling better browser capture. To enable the former, we should add an extensibility point that allows assemblies to define actions that should be performed before, and after, test assembly execution.

To identify this, a new attribute and interface would be defined. The attribute would identify the type that hosts the Before and After methods, and the interface would define the contract for the method signatures.
# Proposed types:
### BeforeAfterAssemblyTestsAttribute

```
class BeforeAfterAssemblyTestsAttribute : Attribute
{
    public Type BeforeAfterAssemblyType { get; }
}
```
### IBeforeAfterAssemblyTests

```
interface IBeforeAfterAssemblyTests
{
    Task Before();
    Task After();
}
```
## Potential Issues
- Should IBeforeAfterAssemblyTests allow a parameterized constructor?
- Should the `IBeforeAfterAssemblyTest` interface simply be merged with `BeforeAfterAssemblyTestsAttribute`? I lean no one this, simply because the assembly might be interested in holding significant state inbetween `Before` and `After` and would allow cleaner code for the end user of the framework.
## Changelog

Changed Before to be parameterless. If we're going to pass it an assembly info, we should do it in the constructor.

**Comment by RichiCoder1** [comment]
Having implemented this, `Before` and `After` aren't the most descriptive of methods. Changing them to `BeforeTestAssemblyStartedAsync` and `AfterTestAssemblyFinishingAsync` respectively. Also aligns them slightly more with how their implemented in `AfterTestAssemblyStarting` and `BeforeTestAssemblyFinished` respectively.

**Comment by RichiCoder1** [comment]
Implemented in 61fd549.
